media/js/products/vpn/geo.js

```diff
@@ -57,54 +57,212 @@
+
+            window.dataLayer.push({
+                'event': 'non-interaction',
+                'eAction': 'vpn-availability',
+                'eLabel': 'variable-price-available'
+            });
+
+            return 'variable-price';
+        }
+
+        // Countries where we should show the waitlist links.
         window.dataLayer.push({
             'event': 'non-interaction',
             'eAction': 'vpn-availability',
             'eLabel': 'not-available'
         });
 
-        return false;
+        return 'not-available';
+    };
+
+    VPN.setAvailability = function(availability, language) {
+        if (availability === 'fixed-price') {
+            // If VPN is available in countries that support fixed pricing,
+            // show the fixed pricing subscribe links.
+            VPN.showFixedPricing();
+        } else if (availability === 'variable-price') {
+            // If VPN is available in countries that support variable pricing,
+            // show the subscription plan options.
+            VPN.showVariablePricing();
+        } else if (availability === 'not-available') {
+            // If VPN is not available in country then show "Join the Waitlist" state.
+            VPN.showJoinWaitList();
+        } else {
+            // If we can't determine someone's country then fall back to page language.
+            // Both /de/ and /fr/ get variable pricing, and the rest get fixed.
+            var lang = document.getElementsByTagName('html')[0].getAttribute('lang') || language;
+            if (lang === 'de' || lang === 'fr') {
+                VPN.showVariablePricing();
+            } else {
+                VPN.showFixedPricing();
+            }
+        }
+    };
+
+    VPN.showJoinWaitList = function() {
+        document.body.classList.add('show-vpn-waitlist');
+        VPN.renderJoinWaitlistButtons();
+    };
+
+    VPN.showFixedPricing = function() {
+        document.body.classList.add('show-vpn-fixed-pricing');
+    };
+
+    VPN.showVariablePricing = function() {
+        document.body.classList.add('show-vpn-variable-pricing');
+        VPN.renderScrollToPricingButtons();
+    };
+
+    VPN.renderScrollToPricingButtons = function() {
+        var template = document.getElementById('scroll-to-pricing-template');
+        var primaryTarget = document.querySelector('.js-target-primary-cta');
+        var secondaryTargets = document.querySelectorAll('.js-target-secondary-cta');
+        var navigationTarget = document.querySelector('.js-target-navigation-cta');
+        var footerTarget = document.querySelector('.js-target-footer-cta');
+        var content = template.content || template;
+        var clone;
+        var button;
+
+        if (!template || !content) {
+            return;
+        }
+
+        if (primaryTarget) {
+            clone = content.querySelector('.js-vpn-variable-pricing').cloneNode(true);
+            button = clone.querySelector('.mzp-c-button');
```

Comment:
Both `clone` and `button` are the same in each `if` statement. Could you set this above in the main declaration block?

Reply:
Hmm I could, but then the individual attributes specific to each button type would all be added together, so I'd need to add some clean up logic after each iteration. With that in mind, I think it's probably simpler to do a fresh clone each time so we can avoid any unwanted side effects.